cns/restserver/ipam.go

```diff
@@ -176,9 +170,12 @@ func (service *HTTPRestService) requestIPConfigsHandler(w http.ResponseWriter, r
 	if service.IPConfigsHandlerMiddleware != nil {
 		// Wrap the default datapath handlers with the middleware
 		wrappedHandler := service.IPConfigsHandlerMiddleware.IPConfigsRequestHandlerWrapper(service.requestIPConfigHandlerHelper, service.releaseIPConfigHandlerHelper)
-		ipConfigsResp, err = wrappedHandler(r.Context(), ipconfigsRequest)
+		ipConfigsResp, err = wrappedHandler(r.Context(), &ipconfigsRequest)
+		if ipconfigsRequest.AddInterfacesDataToResponse {
```

Comment:
now that i have taken a better look, we could have had this field `AddInterfacesDataToResponse` in the `ipConfigsResp` instead of the request. That way we would not have to modify anything related to `ipConfigsRequest`. Plus I think it makes sense that it should be the response's responsibility and not the request to tell CNS to add interfaces data to itself

Reply:
should I introduce a field in PodIpInfo in that case? it doesn't seem right to add this to the current struct ipConfigsResponse, does not fit well the usages & pattern & type of fields in ipconfigsResponse currently

Reply:
yes lets try that

Reply:
yes we should only have one single podIPInfo at a time